dbms/src/DataStreams/WindowTransformAction.h

```diff
@@ -319,6 +321,9 @@ struct WindowTransformAction
 
     bool has_rank_or_dense_rank = false;
 
+    // When preceding and following boundary type are all unbounded we can apply shortcut for agg functions.
+    bool has_shortcut = false;
```

Comment:
maybe we can rename it to `support_batch_calculate` in case in the furture, we will support more batch calculate for some specific window/agg function?

Reply:
> maybe we can rename it to `support_batch_calculate` in case in the furture, we will support more batch calculate for some specific window/agg function?

done